ivy/src/main/scala/sbt/CompatibilityWarning.scala

```diff
@@ -0,0 +1,25 @@
+package sbt
+
+private[sbt] object CompatibilityWarning {
+  def apply(module: IvySbt#Module, mavenStyle: Boolean, log: Logger): Unit = {
+    if (mavenStyle) {
+      processIntransitive(module, log)
+    }
+  }
+
+  def processIntransitive(module: IvySbt#Module, log: Logger): Unit = {
+    val directDependencies: Seq[ModuleID] = module.moduleSettings match {
+      case x: InlineConfiguration             => x.dependencies
+      case x: InlineConfigurationWithExcludes => x.dependencies
+      case _                                  => Seq()
+    }
+    directDependencies foreach { m =>
+      if (!m.isTransitive) {
+        log.warn(
+          s"""Found intransitive dependency ($m), but Maven repositories do not support intransitive dependencies.
```

Comment:
Shall we try and make a reference to the fact that this is dependant on `publishMavenStyle` (as it is in the notes)?

I'm just not sure if "but Maven repositories" is sufficient to understand that this is because `publishMavenStyle` is set to `true`.